apps/aehttp/src/sc_ws_api_jsonrpc.erl

```diff
@@ -292,6 +292,28 @@ process_request(#{<<"method">> := <<"channels.get.contract_call">>,
             end;
         _ -> {error, broken_code}
     end;
+process_request(#{<<"method">> := <<"channels.dry_run.call_contract">>,
+                  <<"params">> := #{<<"contract">>    := ContractE,
+                                    <<"abi_version">> := ABIVersion,
+                                    <<"amount">>      := Amount,
+                                    <<"call_data">>   := CallDataE}}, FsmPid) ->
+    case {aehttp_api_encoder:safe_decode(contract_pubkey, ContractE),
+          bytearray_decode(CallDataE)} of
+        {{ok, Contract}, {ok, CallData}} ->
+            case aesc_fsm:dry_run_contract(FsmPid,
+                                           #{contract     => Contract,
+                                             abi_version  => ABIVersion,
+                                             amount       => Amount,
+                                             call_data    => CallData}) of
+                {ok, Call} ->
+                  {reply, #{ action     => <<"dry_run">>
+                           , tag        => <<"call_contract">>
+                           , {int,type} => reply
+                           , payload    => aect_call:serialize_for_client(Call) }};
```

Comment:
May interact with https://github.com/aeternity/aeternity/pull/2201/files?utf8=%E2%9C%93&diff=split&w=1#diff-89b8ec7c2f28f3e3c7d21ca9bf070bb7R3037 though I see no evident conflict.

Reply:
Rebased on top of `master` to check if there is a hidden dependency